core/src/main/java/com/scalar/db/storage/cosmos/ConcatenationVisitor.java

```diff
@@ -19,88 +19,89 @@
  * @author Yuji Ito
  */
 @NotThreadSafe
-public class ConcatenationVisitor implements ValueVisitor {
-  private final List<String> values;
+public class ConcatenationVisitor implements ColumnVisitor {
+  private final List<String> columns;
 
   public ConcatenationVisitor() {
-    values = new ArrayList<>();
+    columns = new ArrayList<>();
   }
 
   public String build() {
-    // TODO What if the string or blob value includes `:`?
-    return String.join(":", values);
+    // TODO What if the string or blob column includes `:`?
+    return String.join(":", columns);
   }
 
   /**
-   * Sets the specified {@code BooleanValue} to the key string
+   * Sets the specified {@code BooleanColumn} to the key string
    *
-   * @param value a {@code BooleanValue} to be set
+   * @param column a {@code BooleanColumn} to be set
    */
   @Override
-  public void visit(BooleanValue value) {
-    values.add(String.valueOf(value.get()));
+  public void visit(BooleanColumn column) {
+    columns.add(String.valueOf(column.getBooleanValue()));
   }
```

Comment:
Since this class is supposed to be used to create concatenated keys (partition keys and clustering keys), the columns should not be null. So I think we can add an assertion as follows:

```suggestion
  public void visit(BooleanColumn column) {
    assert !column.hasNullValue();
    columns.add(String.valueOf(column.getBooleanValue()));
  }
```

The same applies to the other methods.